src/main/kotlin/no/nav/syfo/client/OppgaveService.kt

```diff
@@ -0,0 +1,170 @@
+package no.nav.syfo.client
+
+import no.nav.helsearbeidsgiver.oppgave.OppgaveClient
+import no.nav.helsearbeidsgiver.oppgave.domain.Behandlingstema
+import no.nav.helsearbeidsgiver.oppgave.domain.Behandlingstype
+import no.nav.helsearbeidsgiver.oppgave.domain.HentOppgaverRequest
+import no.nav.helsearbeidsgiver.oppgave.domain.OppgaveListeResponse
+import no.nav.helsearbeidsgiver.oppgave.domain.Oppgavetype
+import no.nav.helsearbeidsgiver.oppgave.domain.OpprettOppgaveRequest
+import no.nav.helsearbeidsgiver.oppgave.domain.Prioritet
+import no.nav.helsearbeidsgiver.oppgave.domain.Statuskategori
+import no.nav.helsearbeidsgiver.oppgave.domain.Tema
+import no.nav.helsearbeidsgiver.utils.log.sikkerLogger
+import no.nav.syfo.behandling.HentOppgaveException
+import no.nav.syfo.behandling.OpprettFordelingsOppgaveException
+import no.nav.syfo.behandling.OpprettOppgaveException
+import no.nav.syfo.domain.OppgaveResultat
+import no.nav.syfo.util.Metrikk
+import no.nav.syfo.utsattoppgave.BehandlingsKategori
+import java.time.DayOfWeek
+import java.time.LocalDate
+
+class OppgaveService(
+    val oppgaveClient: OppgaveClient,
+    val metrikk: Metrikk,
+) {
+    private val sikkerlogger = sikkerLogger()
+
+    private suspend fun hentOppgave(
+        oppgavetype: String,
+        journalpostId: String,
+    ): OppgaveListeResponse =
+        oppgaveClient
+            .hentOppgaver(
+                HentOppgaverRequest(
+                    oppgavetype = oppgavetype,
+                    tema = Tema.SYK,
+                    statuskategori = Statuskategori.AAPEN,
+                    journalpostId = journalpostId,
+                    sorteringsrekkefolge = "ASC",
+                    sorteringsfelt = "FRIST",
+                ),
+            ).also { sikkerlogger.info("Hentet oppgave for journalpostId {}", journalpostId) }
+
+    private suspend fun hentHvisOppgaveFinnes(
+        oppgavetype: String,
+        journalpostId: String,
+    ): OppgaveResultat? {
+        try {
+            val oppgaveResponse = hentOppgave(oppgavetype = oppgavetype, journalpostId = journalpostId)
```

Comment:
bare litt preferanse om man liker chaining eller ikke, men hva med
` return oppgaveResponse
                .oppgaver
                .firstOrNull()
                ?.id
                ?.let { OppgaveResultat(it, true, false) }`

Reply:
Jeg liker chaining 👯‍♂️  :D